spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/elasticsearch/ElasticsearchRestClientAutoConfigurationTests.java

```diff
@@ -183,6 +183,27 @@ void configureWithCustomPathPrefix() {
 		});
 	}
 
+	@Test
+	void configureWithNoSocketKeepAliveApplyDefaults() {
```

Comment:
This isn't quite what I meant about testing that the defaults match. This checks that Spring Boot's default is `false`, but it doesn't check that Elasticsearch has the same default. Rather than creating an application context and getting the `RestClient` from it, the test itself should use `RestClientBuilder` to create a `RestClient` in its default configuration. It should then check that this `RestClient` has `soKeepAlive` set to `false`.

Reply:
ok, that's resonable. Thanks for your patience~